src/middleware/router.coffee

```diff
@@ -134,13 +139,25 @@ sendRequestToRoutes = (ctx, routes, next) ->
           # on failure
           handleServerError ctx, reason
       else
-        promise = buildNonPrimarySendRequestPromise ctx, route, options, path
+        logger.info "executing non primary: #{route.name}"
+        nonPrimaryPromise = buildNonPrimarySendRequestPromise ctx, route, options, path
+
+
 
       promises.push promise
+      nonPrimaryPromises.push nonPrimaryPromise
 
     (Q.all promises).then ->
       next()
 
+    (Q.all nonPrimaryPromises).then ->
+      messageStore.storeResponse ctx, ->
```

Comment:
Perhaps we should split out the storeReponse function into a storePrimaryResponse function and a storeNonPrimaryResponse function. Then for non primary responses we can just call that and it only adds to the routes array instead of updating the whole message again. Also, this will prevent the transaction status from changing as new routes get completed.

For the transaction status what we actually want is for it to be set to remain as 'Processing' and then only be updated finally once all routes have returned.